src/bindings/python/flux/kvs.py

```diff
@@ -118,10 +118,10 @@ def commit(flux_handle, flags: int = 0) -> int:
     if flux_handle.aux_txn is None:
         return 0
     future = RAW.flux_kvs_commit(flux_handle, None, flags, flux_handle.aux_txn)
-    RAW.flux_future_get(future, None)
+    rv = RAW.flux_future_get(future, None)
     RAW.flux_kvs_txn_destroy(flux_handle.aux_txn)
     flux_handle.aux_txn = None
-    return 0
+    return rv
 
```

Comment:
Should this raise an exception of some sort instead?
(Sorry if that is a dumb question!)

Reply:
Good question.  I kept it in the same style for now b/c other functions that call the KVS C-lib also return -1 on error.

The Python API that currently exists was largely developed based off of the KVS api before KVS transactions existed, before libflux futures existed, and probably other things I'm not remembering.  There's no current need to re-do it, but perhaps it will in time.  Perhaps I'll update #5327 to be a general "modernize Python KVS functions".

Reply:
I was just writing the same question. I think we do need to modernize these functions and use idiomatic Python. My guess is that nobody acutally uses this interface.

Reply:
I don't think any Python developer is going to check the return code of a function for success, so I'd actually say this is borderline a bug. Functions like this should return nothing on success (unless there's a return value) and raise an exception on failure. 

In this case if the `future` here was promoted to a `Future`, then you'd get the exception for free...

Reply:
ya know what, i'm dumb.  I just saw the code that was there and assumed it actually returned -1 on error, but it doesn't.

```Python
handle = flux.Flux()
flux.kvs.put(handle, ".", "bar")
flux.kvs.commit(handle)
```

```
>flux python foo.py
Traceback (most recent call last):
  File "foo.py", line 50, in <module>
    flux.kvs.commit(handle)
  File "/g/g0/achu/chaos/git/flux-framework/flux-core/src/bindings/python/flux/kvs.py", line 121, in commit
    rv = RAW.flux_future_get(future, None)
  File "/g/g0/achu/chaos/git/flux-framework/flux-core/src/bindings/python/flux/wrapper.py", line 220, in __call__
    os.strerror(errnum),
OSError: [Errno 22] EINVAL: Invalid argument
```

```Python
handle = flux.Flux()
flux.kvs.put_mkdir(handle, "")
flux.kvs.commit(handle)
```

```
Traceback (most recent call last):
  File "foo.py", line 49, in <module>
    flux.kvs.put_mkdir(handle, "")
  File "/g/g0/achu/chaos/git/flux-framework/flux-core/src/bindings/python/flux/kvs.py", line 102, in put_mkdir
    return RAW.flux_kvs_txn_mkdir(flux_handle.aux_txn, 0, key)
  File "/g/g0/achu/chaos/git/flux-framework/flux-core/src/bindings/python/flux/wrapper.py", line 220, in __call__
    os.strerror(errnum),
OSError: [Errno 22] EINVAL: Invalid argument
```

Soo I think it's already returning an exception on error.  So perhaps we should remove those "returns" and send an exception.

Reply:
Oh, yeah, I forgot all the wrapped ffi function calls will raise an exception on `< 0` return from the C function...